### PR TITLE
Fix/2.0.11 rc2 integration tests

### LINKED ISSUE #2077: Atlas: stream attachment inventories and attachments
For context https://github.com/blockstack/stacks-blockchain/pull/2013#discussion_r525528001
Screen Shot 2020-11-18 at 5 11 40 PM

### LINKED ISSUE #2569: Speed up `runtime_overflow_unconfirmed_microblocks_integration_test` (currently > 40 minutes)
We should figure out if we can speed up the `runtime_overflow_unconfirmed_microblocks_integration_test` -- it's currently taking over 40 minutes, and it runs as part of our integration tests in Github actions.

### PR DESCRIPTION
**Pull request by jcnelson**
This fixes #2569 and #2572, and adds a new Atlas stress-test that looks for performance regressions that, if passing, will allow us to close #2077 and #2427 (since we will have confirmed that the maximum runtime of these methods is acceptable).

### CODE REVIEW

File: testnet/stacks-node/src/tests/neon_integrations.rs [hunk 1]
```diff
@@ -3886,3 +3892,778 @@ fn antientropy_integration_test() {
...
+            "Failed to mine all transactions: nonces = {:?}, expected {:?} + {}",
+            &account_after_nonces, account_before_nonces, batch_size
+        );
+        panic!();
+    }
+}
+
+#[test]
+#[ignore]
+fn atlas_stress_integration_test() {
```
Comment (kantai) [inline testnet/stacks-node/src/tests/neon_integrations.rs:3949]: This test doesn't appear to work correctly in the Github action.

Comment (lgalabru) [reply testnet/stacks-node/src/tests/neon_integrations.rs:3949]: I think there's an issue line 4630, when retrieving the expected content hash - it looks like the tuple for retrieving the content hash would be invalid.
Do you want me to take over?

File: .github/actions/bitcoin-int-tests/Dockerfile.bitcoin-tests [hunk 2]
```diff
@@ -27,3 +27,4 @@ RUN cargo test -- --test-threads 1 --ignored tests::should_succeed_handling_malf
 RUN cargo test -- --test-threads 1 --ignored tests::neon_integrations::size_overflow_unconfirmed_microblocks_integration_test
 RUN cargo test -- --test-threads 1 --ignored tests::neon_integrations::runtime_overflow_unconfirmed_microblocks_integration_test
 RUN cargo test -- --test-threads 1 --ignored tests::neon_integrations::antientropy_integration_test
+RUN cargo test -- --test-threads 1 --ignored tests::neon_integrations::atlas_stress_integration_test
```
Comment (lgalabru) [inline .github/actions/bitcoin-int-tests/Dockerfile.bitcoin-tests:30]: Can we move this entry to `Dockerfile.atlas-test` instead?

File: testnet/stacks-node/src/tests/neon_integrations.rs [hunk 3]
```diff
@@ -3886,3 +3892,778 @@ fn antientropy_integration_test() {
...
+            );
+        }
+
+        for i in 0..l {
+            if attachments[i] == 0 {
+                continue;
+            }
+            let page_index = attachments[i] % 64; // AttachmentInstance::ATTACHMENTS_INV_PAGE_SIZE;
+            let content_hash = attachment_hashes
+                .get(&(*ibh, page_index))
```
Comment (lgalabru) [inline testnet/stacks-node/src/tests/neon_integrations.rs:4630]: Looking at the insertions, I think it should be:
```suggestion
                .get(&(*ibh, attachments[i]))
```
Will test this patch locally.

Comment (jcnelson) [reply testnet/stacks-node/src/tests/neon_integrations.rs:4630]: Ah, I was wondering why this wasn't working reliably.  I'll try it too -- I want to understand the system better.

Comment (lgalabru) [reply testnet/stacks-node/src/tests/neon_integrations.rs:4630]: Confirming that this test is green on my machine, with this patch.

File: .github/actions/bitcoin-int-tests/Dockerfile.bitcoin-tests (hunk 2 above)
Comment (jcnelson) [reply .github/actions/bitcoin-int-tests/Dockerfile.bitcoin-tests]: Done

File: testnet/stacks-node/src/tests/neon_integrations.rs (hunk 3 above)
Comment (jcnelson) [reply testnet/stacks-node/src/tests/neon_integrations.rs]: Fixed here as well; let's see if the CI takes it.

File: testnet/stacks-node/src/tests/neon_integrations.rs (hunk 1 above)
Comment (jcnelson) [reply testnet/stacks-node/src/tests/neon_integrations.rs:3949]: I think I've fixed it, but let's see if the CI takes it.

**Review by lgalabru — APPROVED** [review]
LGTM, thank you @jcnelson!

**Review by kantai — APPROVED** [review]
LGTM!

**Pull request merged**